### PR TITLE
Ziti dump native api

### PR DESCRIPTION
**Pull request by mary-dcouto**

### CODE REVIEW

File: ZitiNativeApiForDotnetCore/msvc-build.bat [hunk 1]
```diff
@@ -32,15 +32,23 @@ REM copy library/ZitiStatus.cs ../Ziti.NET.Standard/src/OpenZiti
 
 ECHO Build from cmake using: 
 ECHO     cmake --build %BUILDFOLDER%\x86 --config Debug
+cmake --build %BUILDFOLDER%\x86 --config Debug
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/msvc-build.bat]: these were intentionally left out so that people would/could do this on their own. i don't want to invoke this for them. let's revert these two changes

File: ZitiNativeApiForDotnetCore/msvc-build.bat [hunk 2]
```diff
@@ -32,15 +32,23 @@ REM copy library/ZitiStatus.cs ../Ziti.NET.Standard/src/OpenZiti
...
 cmake --build %BUILDFOLDER%\x86 --config Release
 ECHO. 
 ECHO     cmake --build %BUILDFOLDER%\x64 --config Debug
+cmake --build %BUILDFOLDER%\x64 --config Debug
 ECHO     cmake --build %BUILDFOLDER%\x64 --config Release
 cmake --build %BUILDFOLDER%\x64 --config Release
 ECHO. 
 ECHO Or open %BUILDFOLDER%\ziti-sdk.sln
 
+REM install nuget
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/msvc-build.bat]: i considered removing the entire file tbh - but let's revert these changes now too

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c [hunk 3]
```diff
@@ -9,6 +9,8 @@
 #define strdup _strdup
 #endif
 
+#define MAXBUFFERLEN 4096
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/library/ziti4dotnet.c]: make this huge... 64k - there's no need to truncate a line

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c [hunk 4]
```diff
@@ -86,4 +88,67 @@ int ziti_context_event_status(const ziti_event_t* e) {
...
+static int ziti_dump_to_log_op(void* stringsBuilder, const char* fmt, ...) {
+    static char line[4096];
+
+    va_list vargs;
+    va_start(vargs, fmt);
+    vsnprintf(line, sizeof(line), fmt, vargs);
+    va_end(vargs);
+
+    if (strlen(stringsBuilder) + strlen(line) > MAXBUFFERLEN) {
+        return -1;
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/library/ziti4dotnet.c]: why not just append what you can in this case? and if you make the buffer huge hopefully this won't be an issue

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c [hunk 5]
```diff
@@ -86,4 +88,67 @@ int ziti_context_event_status(const ziti_event_t* e) {
...
         return 0;
     }
-}
\ No newline at end of file
+}
+
+typedef int (*printer)(void* arg, const char* fmt, ...);
+
+static int ziti_dump_to_log_op(void* stringsBuilder, const char* fmt, ...) {
+    static char line[4096];
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/library/ziti4dotnet.c]: 4096 --> MAXBUFFERLEN ?

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c [hunk 6]
```diff
@@ -86,4 +88,67 @@ int ziti_context_event_status(const ziti_event_t* e) {
...
+    if (strlen(stringsBuilder) + strlen(line) > MAXBUFFERLEN) {
+        return -1;
+    }
+    // write/append to the buffer
+    strncat(stringsBuilder, line, sizeof(line));
+    return 0;
+}
+
+static int ziti_dump_to_file_op(void* fp, const char* fmt, ...) {
+    static char line[4096];
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/library/ziti4dotnet.c]: again - why not use MAXBUFFERLEN?

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c [hunk 7]
```diff
@@ -86,4 +88,67 @@ int ziti_context_event_status(const ziti_event_t* e) {
...
+    free(buffer);
+}
+
+void z4d_ziti_dump_file(ziti_context ztx, const char* outputFile) {
+    FILE* fp;
+    fp = fopen(outputFile, "w+");
+    if (fp == NULL)
+    {
+        printf("ziti dump to file failed. Unable to Read / Write / Create File");
+        return;
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/library/ziti4dotnet.c:134]: you would need to call fclose here, no?

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c [hunk 8]
```diff
@@ -86,4 +88,67 @@ int ziti_context_event_status(const ziti_event_t* e) {
...
+
+    return 0;
+}
+
+void z4d_ziti_dump_log(ziti_context ztx) {
+    char* buffer;
+    buffer = malloc(MAXBUFFERLEN * sizeof(char));
+    buffer[0] = 0;
+    ziti_dump(ztx, (printer)ziti_dump_to_log_op, buffer);
+    printf("ziti dump to log %s", buffer);
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/library/ziti4dotnet.c:124]: we don't need this printf do we???

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c [hunk 9]
```diff
@@ -86,4 +88,67 @@ int ziti_context_event_status(const ziti_event_t* e) {
...
+    free(buffer);
+}
+
+void z4d_ziti_dump_file(ziti_context ztx, const char* outputFile) {
+    FILE* fp;
+    fp = fopen(outputFile, "w+");
+    if (fp == NULL)
+    {
+        printf("ziti dump to file failed. Unable to Read / Write / Create File");
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/library/ziti4dotnet.c:133]: should this be a log message not printf?

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c [hunk 10]
```diff
@@ -86,4 +88,67 @@ int ziti_context_event_status(const ziti_event_t* e) {
...
+        return;
+    }
+    uv_timeval64_t dump_time;
+    uv_gettimeofday(&dump_time);
+
+    char time_str[32];
+    struct tm* start_tm = gmtime(&dump_time.tv_sec);
+    strftime(time_str, sizeof(time_str), "%a %b %d %Y, %X %p", start_tm);
+
+    fprintf(fp, "Ziti Dump starting: %s\n", time_str);
```
Comment (dovholuknf) [inline ZitiNativeApiForDotnetCore/library/ziti4dotnet.c:143]: shouldn't this be a call to ziti_log too? why not?

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c (hunk 7 above)
Comment (mary-dcouto) [reply ZitiNativeApiForDotnetCore/library/ziti4dotnet.c:134]: This block is called if fp == NULL, so we don't call fclose here.

File: ZitiNativeApiForDotnetCore/msvc-build.bat (hunk 1 above)
Comment (mary-dcouto) [reply ZitiNativeApiForDotnetCore/msvc-build.bat]: done

File: ZitiNativeApiForDotnetCore/msvc-build.bat (hunk 2 above)
Comment (mary-dcouto) [reply ZitiNativeApiForDotnetCore/msvc-build.bat]: done

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c (hunk 5 above)
Comment (mary-dcouto) [reply ZitiNativeApiForDotnetCore/library/ziti4dotnet.c]: done

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c (hunk 6 above)
Comment (mary-dcouto) [reply ZitiNativeApiForDotnetCore/library/ziti4dotnet.c]: done

File: ZitiNativeApiForDotnetCore/library/ziti4dotnet.c (hunk 8 above)
Comment (mary-dcouto) [reply ZitiNativeApiForDotnetCore/library/ziti4dotnet.c:124]: ZITI_LOG is not working in the ziti4dotnet.c, so I used printf statements. I may need some more time to check why ZITI_LOG is not working in native library
